src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -671,9 +671,11 @@ void ProcessOverride (OverrideInformation overrideInformation)
 
 			var isInstantiated = Annotations.IsInstantiated (method.DeclaringType);
 
+			var isStaticInterfaceOverride = @base.IsStatic && method.IsStatic;
+
 			// We don't need to mark overrides until it is possible that the type could be instantiated
 			// Note : The base type is interface check should be removed once we have base type sweeping
-			if (IsInterfaceOverrideThatDoesNotNeedMarked (overrideInformation, isInstantiated))
+			if (!isStaticInterfaceOverride && IsInterfaceOverrideThatDoesNotNeedMarked (overrideInformation, isInstantiated))
 				return;
 
 			if (!isInstantiated && !@base.IsAbstract && _context.IsOptimizationEnabled (CodeOptimizations.OverrideRemoval, method))
```

Comment:
(Can't comment on code below this line directly):
Naming consideration: Currently all of this code uses the term 'virtual method' - which is correct for normal virtual methods, it's a slight stretch for interface methods (but they do act like virtual methods for the most part), but now it's a rather larger stretch for static interface methods - those are definitely no virtual in the typical sense of the word.

Not for this checking, but I think we should consider changing the name to something more descriptive... (although not sure what it should be). At least we need to have a good comment somewhere describing what the logic covers and how it's supposed to work high-level.

Honestly it might be wort a doc in the linker repo. Virtual method and interface marking is one of the most complex behaviors in the linker.

Reply:
Ended up not using this path as you suggested and relying on MapVirtualMethods()'s MapOverrides (after by passing the short circuit for the static method check). All static overrides are done exclusively via [MethodImpl ](https://github.com/dotnet/runtime/blob/main/docs/design/specs/Ecma-335-Augments.md#ii1032-the-override-directive)records. (Which translates to .override directives)